docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,244 @@
+    if (buttonPress == nil) { return; }
+
+    NSDictionary *userInfo = @{@"sdlManager" : self.sdlManager};
+    [[NSNotificationCenter defaultCenter] postNotificationName:<#Notification Name#> object:nil userInfo:managers];
+}];
+self.sdlManager.screenManager.softButtonObjects = @[softButton];
+```
+
+##### Swift
+```swift
+let softButton = SDLSoftButtonObject(name: "Button", state: SDLSoftButtonState(stateName: "State", text: "Data", artwork: nil), handler: { (buttonPress, butonEvent) in
+    guard buttonPress == nil else { return }
+    
+    let userInfo = ["sdlManager":self.sdlManager]
+    NotificationCenter.default.post(name: NSNotification.Name(rawValue: <#Notification Name#>), object: nil, userInfo: managers)
+})
+self.sdlManager.screenManager.softButtonObjects = [softButton];
+```
+### Create the EventEmitter Class
+
+Create the new class and add a listener for the notificaiton and all required methods for sending an event.
+
+##### Objective-C
+###### SDLEventEmitter.h
+```objc
+#import <React/RCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLEventEmitter : RCTEventEmitter
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+###### SDLEventEmitter.m
+```objc
+#import "SDLEventEmitter.h"
+#import "ProxyManager.h"
+#import <React/RCTConvert.h>
+#import <SmartDeviceLink/SmartDeviceLink.h>
+
+@implementation SDLEventEmitter
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init {
+    self = [super init];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(getDoActionNotification:) name:<#Notification Name#> object:nil];
+
+    return self;
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+    return @[@"DoAction"];
+}
+
+- (void)getDoActionNotification:(NSNotification *)notification {
+    if(self.sdlManager == nil) {
+        self.sdlManager = notification.userInfo[@"sdlManager"];
+    }
+    [self sendEventWithName:@"DoAction" body:@{@"type": @"actionType"}];
+}
+
+@end
+```
+
+##### Swift
+```swift
+@objc(SDLEventEmitter)
+class SDLEventEmitter: RCTEventEmitter {
+
+override init() {
+    NotificationCenter.default.addObserver(self, selector: #selector(doAction(_:)), name: Notification.Name(rawValue: "<#Notification Name#>", object: nil)
+    super.init()
+}
+
+@objc func doAction(_ notification: Notification) {
+    if self.sdlManger == nil {
+        self.sdlManager = notification.userInfo["sdlManager"]
+    }
+
+    sendEvent(withName: "DoAction", body: ["type": "actionType"])
+}
+
+override func supportedEvents() -> [String]! {
+    return ["DoAction"]
+}
+
+}
+```
+
+!!! NOTE
+Make sure you add `#import "React/RCTEventEmitter.h"` to the apps bridging header before moving forward if you're making a Swift application.
+!!!
+
+!!! NOTE
+If you're using Swift, you will need to create the Objective-C bridging class for `SDLEventEmitter` and add the proper `RCT_EXTERN_METHOD` wrapper.
```

Comment:
What does this mean? We don't want to put two notes right next to each other, can we combine them or move one of them elsewhere? Or the second one doesn't look like it has to be a note, it looks like it's describing the below code?